Mathlib/Data/Nat/Digits.lean

```diff
@@ -516,6 +517,59 @@ lemma self_div_pow_eq_ofDigits_drop (i n : ℕ) (h : 2 ≤ p):
     (fun l hl ↦ digits_lt_base h hl)
   exact (ofDigits_digits p n).symm
 
+open BigOperators
+
+theorem sub_one_mul_sum_div_pow_eq_sub_sum_digits' (L : List ℕ) (w₁ : ∀ (l : ℕ), l ∈ L → l < p)
+    (w₂ : ∀ (h : L ≠ []), L.getLast h ≠ 0) : (p - 1) * ∑ i in Finset.range L.length,
+    (ofDigits p L) / p ^ i.succ = (ofDigits p L) - L.sum := by
+  obtain h | h | h : 1 < p ∨ 1 = p ∨ p < 1 := trichotomous 1 p
```

Comment:
If you use this trick then you won't have to rewrite by `← h` in the middle case below.
```suggestion
  obtain h | rfl | h : 1 < p ∨ 1 = p ∨ p < 1 := trichotomous 1 p
```

Reply:
(same comment applies in the proof of the corollary below)